lightning/src/onion_message/messenger.rs

```diff
@@ -344,6 +347,7 @@ fn packet_payloads_and_keys<T: secp256k1::Signing + secp256k1::Verification>(
 		} else if let Some(encrypted_payload) = enc_payload_opt {
 			payloads.push((Payload::Receive {
 				control_tlvs: ReceiveControlTlvs::Blinded(encrypted_payload),
+				reply_path: reply_path.take(),
```

Comment:
IIUC, `reply_path` is captured by reference (I think?) in the closure, so it will be `None` if this case is hit more than once when called by `construct_keys_callback`. Is this the desired behavior? I guess so because there would only be one `Payload::Receive`?

Related, the same may occur below where `reply_path.take()` is called again outside of the closure. Partly trying to check my understanding if that case will be hit below but not in the closure.

Reply:
Yep, that's intentional. There should only be one `Payload::Receive` pushed, so `reply_path` should only be `take`n once.